src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -0,0 +1,169 @@
+
+        std::mutex m_mutex;
+        std::tuple<std::optional<SourceValueType>, std::optional<OtherValueTypes>...> m_values{};
+        size_t m_completed_count{0};
+
+        const std::size_t m_total_completed_count = 1 + sizeof...(OtherValueTypes);
+
+        bool has_values_for_observer()
+        {
+            return std::apply([](const auto&... cached_values)
+            {
+                // Variadic expansion used
+                return (cached_values.has_value() && ...);
+            }, m_values);
+        }
+
+        auto combine_cached_values()
+        {
+            return std::apply([&](const auto&... cached_values)
+            {
+                // Variadic expansion used
+                return m_combiner(cached_values.value()...);
+            }, m_values);
+        }
+    };
+
+    /**
+     * \brief Templated helper function for subscribing to variadic 'other' observables.
+     *
+     * \param observables is a variadic tuple. The first observable is the current observable and rest are the other observables.
+     * \param subscriber is the downstream subscriber.
+     * \param coordinator manages the cache of emission from the observables and coordinate dispatching.
+     */
+    template<size_t...I>
+    static void subscribe_other_observables(std::index_sequence<I...>, // Used in compile time for variadic expansion
+                                            const auto& observables,
+                                            const auto& subscriber,
+                                            const auto& coordinator)
+    {
+        // +1 because the first element in tuple is the current observable, and you want to subscribe to the 'other' observables.
+        // (Use variadic expansion to iterate the observables)
+        (subscribe_observable<I + 1>(std::get<I>(observables), subscriber, coordinator), ...);
+    }
+
+    template<size_t I, constraint::observable TObs>
+    static void subscribe_observable(const TObs& observable, const auto& subscriber, const auto& coordinator)
+    {
+        using ValueType = utils::extract_observable_type_t<TObs>;
+        observable.subscribe(create_subscriber_with_state<ValueType>(
+            subscriber.get_subscription().make_child(),
+            subscriber,
+            [coordinator](auto&& value, const auto& subscriber)
+            {
+                coordinator->template offer_next_value<I>(std::forward<decltype(value)>(value), subscriber);
+            },
+            utils::forwarding_on_error{},
+            [coordinator](const auto& subscriber)
+            {
+                coordinator->offer_completed(subscriber);
+            }));
+    }
+
+public:
+    template<constraint::subscriber_of_type<DownstreamType> TSub>
+    auto operator()(TSub&& subscriber) const
+    {
+        // TODO: Avoid copying the combiner?
+        auto coordinator = std::make_shared<combine_latest_coordinator_t<Type, utils::extract_observable_type_t<TObservable>...>>(m_combiner);
```

Comment:
@victimsnino I make `combine_latest_coordinator_t` private class to this class so that it's hidden from the outer world.